### DISCUSSION TITLE
Best practices for using Smart Connections + Obsidian

### DISCUSSION THREAD

**Discussion by trh3** [brianpetro/obsidian-smart-connections#218] [Q&A]
Hello! 

I've just been toying around with Smart Connections, and it is a very interesting tool! I am fairly new to using Obsidian itself and was wondering if anybody had any tips/tricks/best practices for structuring one's Vault so that Smart Connections can make best use of it. Thought I'd make this a Discussion so the community can take advantage of it!

Thanks!

**Comment by QuinnHerden** [comment]
I would love to be see a larger discussion!

My small bit of advice would be to have a folder like my "_" folder.
This is where I store utility files.

I keep templates and views (from the dataview plugin) and many other files that aren't really notes, but help run my vault.

I've added this folder to the exclude list on Smart Connections.

**Comment by zeroliu** [comment]
I am also interested to hear suggestions to shape notes to work better with the plugin. 

If the current embedding strategy is based on 'block's defined by headings, using lists vs headings can play an important role here. I'm curious whether the embedding works better with more texts or more atomic content. 

Also, since ada-v2 has a limitation of 8191 tokens. I wonder what would happen if a block of text exceeds the limit. 

Many of these suggestions could change if the plugin adopts different models or changes the embedding strategy. @brianpetro do you plan to make any changes in this area in the near future?

**Reply by brianpetro** [reply]
> whether the embedding works better with more texts or more atomic content.

It depends.

> Many of these suggestions could change if the plugin adopts different models

My focus is on model-agnostic features. Future update will include "normalizing" the use of these models, meaning that integration of additional models will simply require mapping to the "normalized" properties used by Smart Connections.

> changes the embedding strategy

I have plans for improving the embedding strategy. So far, these changes are things that should be toggle-able in the settings.

> changes in this area in the near future?

A major version update is in the works.

**Reply by QuinnHerden** [reply]
> If the current embedding strategy is based on 'block's defined by headings, using lists vs headings can play an important role here. I'm curious whether the embedding works better with more texts or more atomic content.

I use Readwise. I automatically export my highlights into an Obsidian directory, using their connector. The format I've chosen outputs a book's highlights under a single heading, then a bullet per highlight.

It appears that this approach prohibits "Smart Connections Chat" from seeing continuity between adjacent bullets.

e.x.

Let's say I have a highlight that says "3 steps to giving an example", then separate highlights for each of the three steps:

```
# Notes & Highlights
- 3 steps to giving an example
- 1. Think of a topic
- 2. Fail to be creative
- 3. Get meta
```

Chat would not be able to adequately answer "What are the 3 steps to giving an example?"

**Reply by brianpetro** [reply]
Currently, it depends how many other/extraneous bullet points are under the same heading. The more there are the less likely it will succeed.

One of my high-priority updates is better parsing of those lists. For example, if there are three or more children items, then threat the parent list item as a heading. So that would solve your example.

🌴
